[OV-36] feat(user): 회원 controller 구현 (#88)

challduck opened the pull request:
## 변경 타입
- [x] 신규 기능 추가/수정
- [ ] 버그 수정
- [ ] 리팩토링
- [ ] 설정
- [ ] 비기능 (주석 등 기능에 영향을 주지 않음)

## 테스트 항목
- [ ] 코드가 의도한 대로 동작하는지 테스트 완료
- [x] 기존 기능에 영향을 주지 않음
- [ ] 관련 문서를 업데이트했거나 업데이트가 필요하지 않음

## 변경 내용
- **as-is**
    - (변경 전 설명을 여기에 작성)

- **to-be**
    - 회원 Controller API 구현
    - API 성공 메세지 분리 구현

## 참조
[OV-36](https://challduck.atlassian.net/browse/OV-36)

## 코멘트
- resolve #88

challduck committed:
[OV-36] feat(user): 회원 controller 구현 (#88)

- 회원 Controller API 구현
- API 성공 메세지 분리 구현

Sunro1994 commented on user/src/main/java/com/owlexpress/user/presentation/UserController.java:
```diff
@@ -0,0 +1,136 @@
+                .data(getUserInfoResponseDto)
+                .build();
+
+        return ResponseEntity.ok().build();
```
commonDto가 갈 곳을 잃었습니다... body에 넣어주세요~~ ㅋㅋㅋㅋㅋ

challduck replied:
확인했습니다! 수정해서 반영하겠습니다~

Sunro1994 commented on user/src/main/java/com/owlexpress/user/presentation/UserController.java:
```diff
@@ -0,0 +1,136 @@
+                .message(DELETE_USER_SUCCESS)
+                .build();
+
+        return ResponseEntity.status(HttpStatus.ACCEPTED).build();
```
이 친구도 body에게 안기게 해주세요

Sunro1994 reviewed (approved):
LGTM!

cnxw4570123 commented on user/src/main/java/com/owlexpress/user/presentation/dto/ApiResponseMessageConstant.java:
```diff
@@ -0,0 +1,11 @@
+package com.owlexpress.user.presentation.dto;
+
+public class ApiResponseMessageConstant {
```
메시지 상수도 따로 빼놓으니까 되게 깔끔하네요 👍

challduck replied:
내용이 별로 없을때는 괜찮은데 하나 둘 늘어가면 관리하기가 어려워지고 응답값이 통일성이 없어지는것 같아서 적용시켜봤습니다!

ggumi030 commented on user/src/main/java/com/owlexpress/user/presentation/UserController.java:
```diff
@@ -0,0 +1,136 @@
+    private final UserService userService;
+
+    @PostMapping("/sign-up")
+    public ResponseEntity<CommonDto<Object>> signUp(
```
전 Object말고 실제 객체타입을 적어주었는데 이거 표기하는 방식도 통일하면 좋을 것 같아요

challduck replied:
실제 응답객체 / Void 타입 둘 중 하나만 되도록 수정하겠습니다!

ggumi030 commented on user/src/main/java/com/owlexpress/user/presentation/UserController.java:
```diff
@@ -0,0 +1,136 @@
+
+    @PostMapping("/sign-up")
+    public ResponseEntity<CommonDto<Object>> signUp(
+            @RequestBody UserSignupRequestDto userSignupRequestDto
```
dto에 valid 처리를 해주셨던걸로 기억하는데 검증은 Service단에서 이루어지나요 ?

challduck replied:
valid 어노테이션 추가 후 적용하도록 수정하겠습니다!

challduck committed:
[OV-36] refactor(user): 회원 controller 구현 (#89)

- ResponseEntity 내 commonDto 응답값 추가
- CommonDto Type Response 객체 및 Void 반환 하도록 수정
